### ISSUE TITLE
Enhancement: allowEmpty - position of empty item

### ISSUE THREAD

**Issue by MiMo42** [marmelab/react-admin#3935]
**What you were expecting:**
SelectInput and AutocompleteInput should both render empty item at top of list.

**What happened instead:**
SelectInput renders empty item at top, AutocompleteInput renders empty item an end of list of suggestions (needs us to scroll).

**Other information:**
From a UI/UX perspective it should be consistent. 

**Environment**

* React-admin version: 16.10
* React version: 3.0.0-beta.5
* Browser: Chrome

**Comment by MiMo42** [comment]
After some time working on this, I decided to omit allowEmpty and emptyText support as a whole.
I came up with the idea, that autocomplete should _always_ allowEmpty (delete text and press reset icon button). Its more user centric and more like other AC inputs out there.
I think the user should always be able to clear and restart to search for matches.
We have form validation to force the user to finally fill out the AutocompleteInput component if necessary.
Unfortunately its a BC break, so no pull request into the official next branch in sight.
For now, i will add the components as custom components to my project.

https://github.com/MiMo42/react-admin/tree/AutocompleteFixes
I modified: 

- AutocompleteInput.tsx
- AutocompleteInput.spec.tsx, 
- useSuggestions.ts,
- useSuggestions.spec.ts, 
- AutocompleteArrayInput.tsx 

Perhaps it is a starting point for others.
